modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2142,10 +2133,60 @@ private void _deleteFileEntries(
 				continue;
 			}
 
-			try {
-				_dlFileEntryLocalService.deleteFileEntry(
+			List<Long> orphanedFileEntryIds = new ArrayList<>();
+
+			if (objectField.isLocalized()) {
+				Map<String, Serializable> oldLocalizedValues =
+					(Map<String, Serializable>)oldValues.get(
+						objectField.getI18nObjectFieldName());
+
+				if (oldLocalizedValues == null) {
+					continue;
+				}
+
+				Map<String, Serializable> newLocalizedValues =
+					(Map<String, Serializable>)newValues.getOrDefault(
+						objectField.getI18nObjectFieldName(),
+						(Serializable)Collections.emptyMap());
+
+				for (Map.Entry<String, Serializable> entry :
+						oldLocalizedValues.entrySet()) {
+
+					if (Objects.equals(
+							entry.getValue(),
+							newLocalizedValues.get(entry.getKey()))) {
+
+						continue;
+					}
+
+					orphanedFileEntryIds.add(
+						GetterUtil.getLong(entry.getValue()));
+				}
+			}
+			else {
+				String objectFieldName = objectField.getName();
+
+				if (Objects.equals(
+						GetterUtil.getLong(newValues.get(objectFieldName)),
+						GetterUtil.getLong(oldValues.get(objectFieldName)))) {
+
+					continue;
+				}
+
+				orphanedFileEntryIds.add(
 					GetterUtil.getLong(oldValues.get(objectFieldName)));
 			}
+
+			try {
+				for (Long orphanedFileEntryId : orphanedFileEntryIds) {
+					if (orphanedFileEntryId == 0) {
+						continue;
+					}
+
+					_dlFileEntryLocalService.deleteFileEntry(
+						orphanedFileEntryId);
```

Comment:
Do you know if we need this `IF`? Can't we use `GetterUtil` and inline it?

Reply:
If we pass the value directly to `deleteFileEntry` and its zero, it will throw an exception, since there are no files with ID zero. Did you think of another way to avoid that?